app/commands/v2/publish.rb

```diff
@@ -27,6 +32,28 @@ def validate!
         ) unless update_type.present?
       end
 
+      def validate_version_lock!
+        current_content_item = DraftContentItem.find_by(content_id: content_id)
+        current_version = Version.find_by(target: current_content_item)
+        previous_version_number = payload[:previous_version]
+
+        return unless current_content_item && current_version && previous_version_number
+
+        raise CommandError.new(
+          code: 409,
+          message: "Conflict",
+          error_details: {
+            error: {
+              code: 409,
+              message: "Version conflict",
+              fields: {
+                previous_version: ["does not match"],
+              }
+            }
+          }
+        ) unless current_version.number == previous_version_number
```

Comment:
Having a multi-line block with an `unless` on the end isn't the easiest to follow.  Would be easier to follow if this used the block form of `unless`.